### ISSUE TITLE
DateToolsSwift - Swift Compiler Error - XCode 9.3 (beta)

### ISSUE THREAD

**Issue by MattCarr147** [MatthewYork/DateTools#244]
I'm unable to compile my project due to the following error:

`.../Pods/DateToolsSwift/DateToolsSwift/DateTools/TimePeriodGroup.swift:105:28: Cannot convert return expression of type '[ArraySlice<TimePeriodProtocol>]' to return type '[AnySequence<TimePeriodProtocol>]'`

This has only occurred after installing the XCode 9.3 beta. 

EDIT: Note that I am using DateToolsSwift v2.0.3

**Comment by tomhartnett** [comment]
I also ran into this on Xcode beta.  Until it's fixed, I've temporarily altered the function so I can continue developing in Xcode 9.3.

I didn't test that the code works per se... but it compiles.

```
public func split(maxSplits: Int, omittingEmptySubsequences: Bool, whereSeparator isSeparator: (TimePeriodProtocol) throws -> Bool) rethrows -> [AnySequence<TimePeriodProtocol>] {
        let slices = try periods.split(maxSplits: maxSplits, omittingEmptySubsequences: omittingEmptySubsequences, whereSeparator: isSeparator)
        var results: [AnySequence<TimePeriodProtocol>] = []
        for slice in slices {
            let result: AnySequence<TimePeriodProtocol> = AnySequence<TimePeriodProtocol>(slice)
            results.append(result)
        }
        return results
    }
```

**Comment by hartbit** [comment]
Or some susinctly:

```
public func split(maxSplits: Int, omittingEmptySubsequences: Bool, whereSeparator isSeparator: (TimePeriodProtocol) throws -> Bool) rethrows -> [AnySequence<TimePeriodProtocol>] {
    return try periods.split(maxSplits: maxSplits, omittingEmptySubsequences: omittingEmptySubsequences, whereSeparator: isSeparator).map(AnySequence.init)
}
```

**Comment by JohnnyTseng** [comment]
I've made a fork of it using @hartbit answer,
to people who uses CocoaPods to install DateTools, that are unable to change the source code directly.

Use
`pod 'DateToolsSwift', :git => 'git@github.com:JohnnyTseng/DateTools.git', :branch => 'xcode-9.3'`
for a temporarily fix.

**Comment by MatthewYork** [comment]
Merged and pod 4.0.0 issued!

**Comment by AlexFlyce** [comment]
If you can't update the pod for any reason, also adding the line: ```.map(AnySequence.init)``` at the return works. (and it's basically what the update is about)
Complete return statement:
```
return try periods.split(maxSplits: maxSplits, omittingEmptySubsequences: omittingEmptySubsequences, whereSeparator: isSeparator).map(AnySequence.init)
```
